Readme.org

```diff
@@ -49,12 +49,43 @@ Concrete example:
 $ cat my-repos.txt | /path/to/tomono
 #+end_src
 
-That should be all ✅.
-
 #+begin_comment
 Yes #uselessuseofcat but it is clearer than < > ! # $) &&*!&♨±⌘︎ to newbies.
 #+end_comment
 
+You are also recommended to fetch all git remotes in new repository to update your local state. 
```

Comment:
I don't think I understand this part--the script just did a full fetch, there should be no new changes incoming after running the script. After a while, yes, but not after you just merged. Does this actually change something for you?

Reply:
See https://github.com/hraban/tomono/blob/4ca8975cdde05b04f657d6f491caae4a83b10b25/tomono#L64

Reply:
Thinking about it I am also confused why this step seems to be required. It should have fetched all the objects for the origins as part of the merging process and in theory should be up to date. As part of my personal testing the origins have not actually changed during the merge process either. I can however confirm it is the most likely cause of issues such as #21. My best guess is that the git repo thinks it is out of sync with the remotes.

Reply:
It also seems to be specific to using remote repositories, if you use full git paths instead of local paths you can replicate the issues trying to rebase merged branches. I believe it should be easy to replicate but I can attempt to make some reproduction steps as well.

Your repos list has to use the full git urls such as the usage example. It seems like there are subtle differences in behavior between what happens when you use the full url vs merge local repositories such as your test scripts.

```
$ cat my-repos.txt
git@github.com:mycompany/my-repo-abc.git  abc
git@github.com:mycompany/my-repo-def.git  def
git@github.com:mycompany/my-lib-uuu.git   uuu  lib/uuu
git@github.com:mycompany/my-lib-zzz.git   zzz  lib/zzz
https://gitee.com/shijie/zhongguo.git     中国
```

Reply:
Interesting, although I should say the integration tests do use local paths: https://tomono.0brg.net/#tests and those seem to reinforce the documented behavior. If there is a difference there that would definitely be a bug of course.

thanks for hanging in there :) I'm still trying to reproduce the exact issue

Reply:
Just for the record: are we talking about first doing a local checkout of a remote branch, and then using _that_ as the remote for the actual tomono action? Because that is unsupported, and it would definitely explain the discrepancy. You wouldn't be the first person to bring that up so it's worth me adding a warning about this to the docs, regardless.

Reply:
Sorry for the delay in response. I am running it similarly to the examples to avoid needing a local clone of the repository. I don't have a good set of example repositories to test with currently but Ill try to see if I can track a few down. The text file I use is something along the lines of:

```txt
git@github.com:org/common-package.git Common
git@github.com:org/sdkA.git metrics Sdk/A
git@github.com:org/sdkb.git auth Sdk/B
```

So I always pull the full repository every run. From what I know about Git I can't think of why this would be an issue when a fetch is part of the script. I just know that in my testing with several very merge heavy repositories it seems like Git is convinced it is missing some history after a merge.